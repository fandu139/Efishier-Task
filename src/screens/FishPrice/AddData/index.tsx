import React, { useEffect } from 'react';
import { View, StyleSheet, Button, ToastAndroid } from 'react-native';
import {
  TextField,
} from 'react-native-material-textfield-plus';
import Colors from '../../../theme/colors';
import { goBack } from '../../../helper/navigation';
import SteinStore from 'stein-js-client';
import { StackScreenProps } from '@react-navigation/stack';
import uuid from 'react-native-uuid';
import moment from 'moment';

const storeArea = new SteinStore(
  'https://stein.efishery.com/v1/storages/5e1edf521073e315924ceab4/option_area',
);
const storeSize = new SteinStore(
  'https://stein.efishery.com/v1/storages/5e1edf521073e315924ceab4/option_size',
);

const store = new SteinStore(
  'https://stein.efishery.com/v1/storages/5e1edf521073e315924ceab4/list',
);

type RootStackParamList = {
  AddDataScreen: {
    typeAction: string,
    item: {
      area_kota: string,
      area_provinsi: string,
      komoditas: string,
      price: string,
      size: string,
      tgl_parsed: string,
      timestamp: string,
      uuid: string,
    },
  };
};

type Props = StackScreenProps<RootStackParamList, 'AddDataScreen'>;

const AddDataScreen: React.FC<Props> = ({ navigation, route }: Props) => {
  const item = route?.params?.item;
  const typeAction = route?.params?.typeAction;
  const [uid, setUid] = React.useState(item?.uuid);
  const [komoditas, setKomoditas] = React.useState(item?.komoditas);
  const [province, setProvince] = React.useState(item?.area_provinsi);
  const [city, setCity] = React.useState(item?.area_kota);
  const [size, setSize] = React.useState(item?.size);
  const [price, setPrice] = React.useState(item?.price);

  useEffect(() => {
    navigation.setOptions({
      title: typeAction === 'add' ? 'Tambah Data' : 'Edit Data',
    });
  });

  const saveData = async (action: string) => {
    if (action === 'add' ) {
      store
        .append('', [
          {
            uuid: uuid.v4(),
            komoditas,
            area_provinsi: province.toUpperCase(),
            area_kota: city.toUpperCase(),
            size,
            price,
            tgl_parsed: moment().format("YYYY-MM-DD"),
            timestamp: moment().unix()
          },
        ])
        .then(() => {
          ToastAndroid.show('Data berhasil di simpan', ToastAndroid.SHORT);
          goBack();
        });
    }

    if (action === 'edit') {
      store
        .edit('', {
          search: { uuid : uid },
          set: {
            komoditas,
            area_provinsi: province.toUpperCase(),
            area_kota: city.toUpperCase(),
            size,
            price,
            tgl_parsed: moment().format("YYYY-MM-DD"),
            timestamp: moment().unix()
          },
        })
        .then(res => {
          ToastAndroid.show('Data berhasil di ubah', ToastAndroid.SHORT);
          goBack();
        });
    }

    if (action === 'delete') {
      store
        .delete('', {
          search: { uuid : uid, komoditas, },
        })
        .then(res => {
          ToastAndroid.show('Data berhasil di hapus', ToastAndroid.SHORT);
          goBack();
        });
    }
  };

  return (
    <View style={styles.svContentContainer}>
      <View style={styles.inputContainer}>
        <TextField
          onChangeText={setKomoditas}
          value={komoditas}
          label="Komoditas"
        />
        <TextField
          onChangeText={setProvince}
          value={province}
          label="Provinsi"
        />
        <TextField
          onChangeText={setCity}
          value={city}
          label="City"
        />
        <TextField
          onChangeText={setSize}
          value={size}
          label="Size"
          keyboardType="numeric"
        />
        <TextField
          onChangeText={setPrice}
          value={price}
          label="Harga"
          keyboardType="numeric"
        />
        <View style={[styles.actionContainer, typeAction === 'edit' && styles.actionTwoContainer]}>
          {typeAction === 'edit' && (
            <Button
            testID="button-clear-filter"
            onPress={() => saveData('delete')}
            title={'Delete Data'}
            color={Colors.RED}
          />)}
          <Button
            testID="button-filter-order"
            onPress={() => saveData(typeAction)}
            title={'Simpan Data'}
          />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  svContentContainer: {
    flexGrow: 1,
    backgroundColor: Colors.WHITE,
    paddingTop: 15,
  },
  actionContainer: {
    padding: 20,
    borderRadius: 5,
    borderColor: Colors.GRAYEF
  },
  actionTwoContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  inputContainer: {
    paddingHorizontal: 20,
  }
});

export default AddDataScreen;
