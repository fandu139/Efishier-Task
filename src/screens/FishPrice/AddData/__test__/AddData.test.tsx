import AddData from '../index';
import React from 'react';
import { render } from '@testing-library/react-native';

describe('Filter screen test', () => {
  describe('Filter screen snapshot test', () => {
    it('should render correctly', () => {
      const navigation = {
        setOptions: jest.fn(),
        replace: jest.fn(),
      };

      const mockedRoute = {
        params: {
          AddDataScreen: {
            typeAction: 'add',
            item: {
              area_kota: 'Palu',
              area_provinsi: 'Sulawesi Tengah',
              komoditas: 'Ikan Tenggiri',
              price: '10000',
              size: '80',
              tgl_parsed: '2022-02-01',
              timestamp: '11111111',
              uuid: 'abds-asdda-sa1111',
            },
          }
        },
      };

      const tree = render(<AddData navigation={navigation as never} route={mockedRoute as never} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
