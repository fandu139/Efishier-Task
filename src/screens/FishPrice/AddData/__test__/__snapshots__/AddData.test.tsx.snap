// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter screen test Filter screen snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 64,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 16,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.38)",
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "bottom": 0,
                "left": -3,
                "position": "absolute",
                "right": -3,
                "top": -3,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "left": "-100%",
                "paddingLeft": "50%",
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "translateY": 20,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": "200%",
              }
            }
          >
            <Text
              collapsable={false}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 16,
                  "includeFontPadding": false,
                  "lineHeight": 16,
                  "textAlign": "left",
                  "textAlignVertical": "top",
                }
              }
            >
              Komoditas
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={Object {}}
              />
              <TextInput
                autoCapitalize="sentences"
                disableFullscreenUI={true}
                editable={true}
                onBlur={[Function]}
                onChange={[Function]}
                onChangeText={[Function]}
                onContentSizeChange={[Function]}
                onFocus={[Function]}
                selectionColor="rgb(0, 145, 234)"
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "includeFontPadding": false,
                      "margin": 0,
                      "padding": 0,
                      "paddingTop": 0,
                      "textAlign": "left",
                      "textAlignVertical": "top",
                      "top": 2,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, .87)",
                      "fontSize": 16,
                      "height": 24,
                    },
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
            Object {
              "minHeight": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            undefined,
            undefined,
          ]
        }
      />
    </View>
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 64,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 16,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.38)",
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "bottom": 0,
                "left": -3,
                "position": "absolute",
                "right": -3,
                "top": -3,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "left": "-100%",
                "paddingLeft": "50%",
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "translateY": 20,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": "200%",
              }
            }
          >
            <Text
              collapsable={false}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 16,
                  "includeFontPadding": false,
                  "lineHeight": 16,
                  "textAlign": "left",
                  "textAlignVertical": "top",
                }
              }
            >
              Provinsi
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={Object {}}
              />
              <TextInput
                autoCapitalize="sentences"
                disableFullscreenUI={true}
                editable={true}
                onBlur={[Function]}
                onChange={[Function]}
                onChangeText={[Function]}
                onContentSizeChange={[Function]}
                onFocus={[Function]}
                selectionColor="rgb(0, 145, 234)"
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "includeFontPadding": false,
                      "margin": 0,
                      "padding": 0,
                      "paddingTop": 0,
                      "textAlign": "left",
                      "textAlignVertical": "top",
                      "top": 2,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, .87)",
                      "fontSize": 16,
                      "height": 24,
                    },
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
            Object {
              "minHeight": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            undefined,
            undefined,
          ]
        }
      />
    </View>
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 64,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 16,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.38)",
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "bottom": 0,
                "left": -3,
                "position": "absolute",
                "right": -3,
                "top": -3,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "left": "-100%",
                "paddingLeft": "50%",
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "translateY": 20,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": "200%",
              }
            }
          >
            <Text
              collapsable={false}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 16,
                  "includeFontPadding": false,
                  "lineHeight": 16,
                  "textAlign": "left",
                  "textAlignVertical": "top",
                }
              }
            >
              City
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={Object {}}
              />
              <TextInput
                autoCapitalize="sentences"
                disableFullscreenUI={true}
                editable={true}
                onBlur={[Function]}
                onChange={[Function]}
                onChangeText={[Function]}
                onContentSizeChange={[Function]}
                onFocus={[Function]}
                selectionColor="rgb(0, 145, 234)"
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "includeFontPadding": false,
                      "margin": 0,
                      "padding": 0,
                      "paddingTop": 0,
                      "textAlign": "left",
                      "textAlignVertical": "top",
                      "top": 2,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, .87)",
                      "fontSize": 16,
                      "height": 24,
                    },
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
            Object {
              "minHeight": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            undefined,
            undefined,
          ]
        }
      />
    </View>
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 64,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 16,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.38)",
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "bottom": 0,
                "left": -3,
                "position": "absolute",
                "right": -3,
                "top": -3,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "left": "-100%",
                "paddingLeft": "50%",
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "translateY": 20,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": "200%",
              }
            }
          >
            <Text
              collapsable={false}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 16,
                  "includeFontPadding": false,
                  "lineHeight": 16,
                  "textAlign": "left",
                  "textAlignVertical": "top",
                }
              }
            >
              Size
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={Object {}}
              />
              <TextInput
                autoCapitalize="sentences"
                disableFullscreenUI={true}
                editable={true}
                keyboardType="numeric"
                onBlur={[Function]}
                onChange={[Function]}
                onChangeText={[Function]}
                onContentSizeChange={[Function]}
                onFocus={[Function]}
                selectionColor="rgb(0, 145, 234)"
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "includeFontPadding": false,
                      "margin": 0,
                      "padding": 0,
                      "paddingTop": 0,
                      "textAlign": "left",
                      "textAlignVertical": "top",
                      "top": 2,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, .87)",
                      "fontSize": 16,
                      "height": 24,
                    },
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
            Object {
              "minHeight": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            undefined,
            undefined,
          ]
        }
      />
    </View>
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 64,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 16,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.38)",
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "bottom": 0,
                "left": -3,
                "position": "absolute",
                "right": -3,
                "top": -3,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "left": "-100%",
                "paddingLeft": "50%",
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "translateY": 20,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": "200%",
              }
            }
          >
            <Text
              collapsable={false}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 16,
                  "includeFontPadding": false,
                  "lineHeight": 16,
                  "textAlign": "left",
                  "textAlignVertical": "top",
                }
              }
            >
              Harga
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={Object {}}
              />
              <TextInput
                autoCapitalize="sentences"
                disableFullscreenUI={true}
                editable={true}
                keyboardType="numeric"
                onBlur={[Function]}
                onChange={[Function]}
                onChangeText={[Function]}
                onContentSizeChange={[Function]}
                onFocus={[Function]}
                selectionColor="rgb(0, 145, 234)"
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "includeFontPadding": false,
                      "margin": 0,
                      "padding": 0,
                      "paddingTop": 0,
                      "textAlign": "left",
                      "textAlignVertical": "top",
                      "top": 2,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, .87)",
                      "fontSize": 16,
                      "height": 24,
                    },
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
            Object {
              "minHeight": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            undefined,
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#EFEFEF",
            "borderRadius": 5,
            "padding": 20,
          },
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="button-filter-order"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Simpan Data
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
